internal/ingress/annotations/auth/main.go

```diff
@@ -108,11 +109,21 @@ func (a auth) Parse(ing *extensions.Ingress) (interface{}, error) {
 		}
 	}
 
-	name := fmt.Sprintf("%v/%v", ing.Namespace, s)
-	secret, err := a.r.GetSecret(name)
+	sns, sname, err := cache.SplitMetaNamespaceKey(s)
 	if err != nil {
 		return nil, ing_errors.LocationDenied{
-			Reason: errors.Wrapf(err, "unexpected error reading secret %v", name),
+			Reason: errors.Wrap(err, "error reading secret name from annotation"),
+		}
+	}
+
+	if sns == "" {
+		sname = fmt.Sprintf("%v/%v", ing.Namespace, sname)
+	}
+
+	secret, err := a.r.GetSecret(sname)
```

Comment:
Let's say the annotation value is `different-ns/my-secret`, then `sns` will be `different-ns` and `sname` will be `my-secret`. The the call will look like `a.r.GetSecret("my-secret")` - is this going to work? Should it not be `a.r.GetSecret("different-ns/my-secret")`?

Reply:
fixed